File: Backend/app/schemas/client/doctor.py
from typing import Optional

from app.schemas.base import BaseResponseSchema, add_padded_id


@add_padded_id()
class DoctorResponse(BaseResponseSchema):
    department_id: int
    title: Optional[str] = None
    introduction: Optional[str] = None
    padded_id: Optional[str] = None
    first_name: Optional[str] = None
    last_name: Optional[str] = None
    department_name: Optional[str] = None

File: medical-client/src/api/doctor.ts
import request from '../utils/request'

export interface Doctor {
  id: number
  department_id: number
  title?: string
  introduction?: string
  padded_id?: string
  first_name?: string
  last_name?: string
  department_name?: string
  created_at: string
  updated_at: string
}

export function listDoctors(departmentId?: number): Promise<Doctor[]> {
  return request.get('/doctors', { params: { department_id: departmentId } })
}

export function getDoctor(id: number): Promise<Doctor> {
  return request.get(`/doctors/${id}`)
}
